### DISCUSSION TITLE
How to disable autocomplete and autofill to the TextField component?

### DISCUSSION THREAD

**Discussion by Narvalex** [microsoft/fluentui-blazor#637] [Q&A] [answered]
Like the `<input type="text" autocomplete="off">`

**Comment by vnbaaij** [comment]
FluentTextField supports `@attributes=AdditionalAttributes` (in both v2 & v3) so you should be able to just add the attribute in the tag and then it would get rendered.

**Reply by Narvalex** [reply]
![image](https://github.com/microsoft/fluentui-blazor/assets/11648387/72c3fcf0-2bf5-42c2-b140-e3a6549b82b0)
 As you can see, it doesn't apply to the input component, but to the root one. How can I set the attribute to the nested input html element?

**Reply by dvoituron** [reply]
FluentUI Web Components use the [Shadow DOM](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_shadow_DOM) features. You have to set the style using **::part**. 

Example: https://stackoverflow.com/questions/75468932/how-to-override-css-style-of-shadow-dom

**Reply by dvoituron** [reply]
I don't know if you can set an autocomplete attribute using CSS. Maybe using a JS script?

Loot at https://developer.mozilla.org/en-US/docs/Web/API/Element/shadowRoot

**Reply by Narvalex** [reply]
I can try set that with Js script, but wouldn't be easier for future work to just add that to the FluentTextField attribute as well? In the same way as placeholder and other attributes are in the FluentTextField attributes parameters API. It will be more performant than invoking a JS script from WebAssembly just for adding that I guess. If I make a PR for adding that, would it be accepted?

**Reply by dvoituron** [reply, marked as answer]
I pushed this PR #648

**Reply by Narvalex** [reply]
Thanks @dvoituron and @vnbaaij
